zschema/compounds.py

```diff
@@ -192,8 +192,11 @@ def validate(self, value):
             raise DataValidationException("record is not a dict", str(value))
         for subkey, subvalue in value.items():
             if subkey not in self.definition:
-                raise DataValidationException("%s is not a valid subkey of root",
-                                              subkey)
+                port = Port(subkey)
```

Comment:
I agree this fix is silly, but I suggest we merge it and file an issue that we broadly don't handle custom types as keys in the validator.